File: ex10/ex10_05/ex10_05.py
def main():
  color_array = []
  colors = ["red", "green", "blue", "yellow", "cyan"]
  rgbs = [[255, 0, 0], [0, 255, 0], [0, 0, 255], [255, 255, 0], [0, 255, 255]]
  color = 0x000000

  for i, val in enumerate(rgbs): 
    color = rgb_to_uint32(val)
    color_array.append({"color": colors[i], "code": color})
    print("Color: " + colors[i] + ", Code: " + str(hex(color)))
  print("-------------------------------")
  search_color = input("Enter color name to search: ")
  if search_color == "":
    print("No color name provided.")
    return
  for item in color_array:
    if item["color"] == search_color.strip():
      print("Found color in color array: " + item["color"])
      color = item["code"]
      rgb = [0, 0, 0]
      uint32_to_rgb(rgb, color)
      print("Value: R(" + str(rgb[0]) + "), G(" + str(rgb[1]) + "), B(" + str(rgb[2]) + ")")
      break 
  return 0    

def rgb_to_uint32(rgb):
  color = (rgb[0] << 16) | (rgb[1] << 8) | rgb[2]
  return color

def uint32_to_rgb(rgb, color):
  rgb[0] = (color >> 16) & 0xFF
  rgb[1] = (color >> 8) & 0xFF
  rgb[2] = color & 0xFF

if __name__ == "__main__":
  main()

File: ex10/ex10_05/ex10_05.js
import promptSync from 'prompt-sync';
const prompt = promptSync();

function main() {
  const colorArray = new Array();
  const colors = ["red", "green", "blue", "yellow", "cyan"];
  const rgbs = [[255, 0, 0], [0, 255, 0], [0, 0, 255], [255, 255, 0], [0, 255, 255]];

  let color = 0x000000;

  for (let i of rgbs) {
    color = rgbToUint32(i);
    colorArray.push({color: colors[rgbs.indexOf(i)], code: color});
    console.log("Color: " + colors[rgbs.indexOf(i)] + ", Code: 0x" + color.toString(16));
  }
  console.log("--------------------------------");
  let searchColor = "";
  searchColor = prompt("Enter color name to search: ");
  if (searchColor === "") {
    console.log("No color name provided.");
    return;
  }
  const found = colorArray.find(item => item.color === searchColor.trim());
  if (found) {
    console.log("Found color in color array: " + found.color);
    color = found.code;
    let rgb = new Uint8Array(3);
    uint32ToRGB(rgb, color);
    console.log("Value: R(" + rgb[0] + "), G(" + rgb[1] + "), B(" + rgb[2] + ")");
  }
  else {
    console.log("Color not found");
  }
  return 0;
}

function rgbToUint32(rgb) {
  let color = (rgb[0] << 16) | (rgb[1] << 8) | rgb[2];
  return color;
}

function uint32ToRGB(rgb, color) {
  rgb[0] = (color >> 16) & 0xFF;
  rgb[1] = (color >> 8) & 0xFF;
  rgb[2] = color & 0xFF;
}
main();